pkg/operator/staticresource/controller.go

```diff
@@ -0,0 +1,289 @@
+package staticresource
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"strings"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/cloud-credential-operator/pkg/assets/v410_00_assets"
+	"github.com/openshift/cloud-credential-operator/pkg/operator/platform"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/resource/resourcehelper"
+	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceread"
+	log "github.com/sirupsen/logrus"
+	admissionregistrationv1beta1 "k8s.io/api/admissionregistration/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	admissionregistrationclientv1beta1 "k8s.io/client-go/kubernetes/typed/admissionregistration/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	controllerName    = "staticresource"
+	deploymentName    = "cloud-credential-operator"
+	operatorNamespace = "openshift-cloud-credential-operator"
+)
+
+var (
+	staticFiles = []string{
+		"v4.1.0/aws-pod-identity-webhook/sa.yaml",
+		"v4.1.0/aws-pod-identity-webhook/clusterrole.yaml",
+		"v4.1.0/aws-pod-identity-webhook/role.yaml",
+		"v4.1.0/aws-pod-identity-webhook/clusterrolebinding.yaml",
+		"v4.1.0/aws-pod-identity-webhook/rolebinding.yaml",
+		"v4.1.0/aws-pod-identity-webhook/svc.yaml",
+	}
+	templateFiles = []string{
+		"v4.1.0/aws-pod-identity-webhook/deployment.yaml",
+		"v4.1.0/aws-pod-identity-webhook/mutatingwebhook.yaml",
+	}
+	defaultCodecs = serializer.NewCodecFactory(scheme.Scheme)
+	defaultCodec  = defaultCodecs.UniversalDeserializer()
+)
+
+type staticResourceInitialApplyController struct {
+	reconciler *staticResourceReconciler
+}
+
+func (c *staticResourceInitialApplyController) Start(stopCh <-chan struct{}) error {
+	c.reconciler.ReconcileResources()
+	<-stopCh
+	return nil
+}
+
+func Add(mgr manager.Manager, kubeconfig string) error {
+	// Currently the only static assets are AWS only
+	infraStatus, err := platform.GetInfraStatusUsingKubeconfig(mgr, kubeconfig)
+	if err != nil {
+		return err
+	}
+	platformType := platform.GetType(infraStatus)
+	if platformType != configv1.AWSPlatformType {
```

Comment:
Is it accurate to say that in one potential future (not implying that we'd want to support this...), we could have a cluster running on GCP accessing resources in AWS?

Reply:
@dgoodwin made the same point about this code not being generic.  I'm renaming the package to `awspodidentity`. 

And no, I don't see this code being used on any other platform since it is really only useful for assuming IAM roles, which... I guess you could do from instances on another cloud but... I can't see us supporting that.